lib/ansible/modules/clustering/etcd3.py

```diff
@@ -107,9 +160,13 @@ def run_module():
     # supports check mode
     module = AnsibleModule(
         argument_spec=module_args,
-        supports_check_mode=True
+        supports_check_mode=True,
+        required_together=[['client_cert', 'client_key'], ['user', 'password']],
     )
 
+    if module.params['ca_cert'] is None and module.params['client_cert'] is not None:
+        module.fail_json(msg="The 'ca_cert' parameter must be defined when 'client_cert' and 'key_cert' are present.")
```

Comment:
But then this error message is inconsistent with the check in condition.

Reply:
Same as above, due to the `required_together`, the following lines are equivalent:
```python
if module.params['ca_cert'] is None and module.params['client_cert'] is not None:
if module.params['ca_cert'] is None and module.params['client_cert'] is not None and module.params['client_key'] is not None:
if module.params['ca_cert'] is None and (module.params['client_cert'] is not None or module.params['client_key'] is not None):
```